pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -114,22 +114,3 @@ func calculateResourceAllocatableRequest(nodeInfo *schedulernodeinfo.NodeInfo, p
 	}
 	return 0, 0
 }
-
-// calculatePodResourceRequest returns the total non-zero requests. If Overhead is defined for the pod and the
-// PodOverhead feature is enabled, the Overhead is added to the result.
-func calculatePodResourceRequest(pod *v1.Pod, resource v1.ResourceName) int64 {
-	var podRequest int64
-	for i := range pod.Spec.Containers {
-		container := &pod.Spec.Containers[i]
-		value := schedutil.GetNonzeroRequestForResource(resource, &container.Resources.Requests)
```

Comment:
In the `GetNonzeroRequestForResource ` function of scheduler package,  if the podSpec doesn't set the CPU and MEM resource request, it will give it a default value

```
// For each of these resources, a pod that doesn't request the resource explicitly
// will be treated as having requested the amount indicated below, for the purpose
// of computing priority only. This ensures that when scheduling zero-request pods, such
// pods won't all be scheduled to the machine with the smallest in-use request,
// and that when scheduling regular pods, such pods will not see zero-request pods as
// consuming no resources whatsoever. We chose these values to be similar to the
// resources that we give to cluster addon pods (#10653). But they are pretty arbitrary.
// As described in #11713, we use request instead of limit to deal with resource requirements.
const (
	// DefaultMilliCPURequest defines default milli cpu request number.
	DefaultMilliCPURequest int64 = 100 // 0.1 core
	// DefaultMemoryRequest defines default memory request size.
	DefaultMemoryRequest int64 = 200 * 1024 * 1024 // 200 MB
)
```
But in the util function called `GetResourceRequest` of `k8s.io/kubernetes/pkg/api/v1/resource` package, it doesn't consider this scenario. I think that is the reason why the bazel-test failed.

I am not sure if the default request value of  CPU and MEM is a general feature for every pod. 
IMO, we may need to adapt the `default value logic` to the  `GetResourceRequest` function. Because every pod need to be scheduled by kube-scheduler.

Reply:
@Huang-Wei @smarterclayton WDYT?

Reply:
If we need to modify the util function `GetResourceRequest`, Please tell me if I need to do it in a new PR.

Reply:
> In the GetNonzeroRequestForResource function of scheduler package, if the podSpec doesn't set the CPU and MEM resource request, it will give it a default value

Intents are explained here:

https://github.com/kubernetes/kubernetes/blob/fe2fdcd6953a877664cbdb784888c0e7f636ce40/pkg/scheduler/util/non_zero.go#L24-L31

So let's respect that, but that's not "default" behavior - which is used only for Scoring.

> If we need to modify the util function GetResourceRequest, Please tell me if I need to do it in a new PR.

Nope. GetResourceRequest should be a canonical behavior.